### PR TITLE
fix(Dgraph): backup API should not keep the client waiting

### PR DESCRIPTION
**Pull request by ajeetdsouza**
Fixes DGRAPH-2540.

Backups and exports are long-running operations, and we end up making the client wait till the client completes. This creates problems - not only does it end up keeping an unused connection open for long periods of time, but also causes HTTP timeouts when the operation takes too long (~ 50 GB of data).

This PR changes the backup/export API to return immediately after queueing the backup, and allows the user to see the status of the last backup/export via the /health endpoint.

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7538)

### CODE REVIEW

File: worker/backup_ee.go [hunk 1]
```diff
@@ -180,13 +193,16 @@ func ProcessBackupRequest(ctx context.Context, req *pb.BackupRequest, forceFull
 	predMap := make(map[uint32][]string)
 	for gid, group := range state.Groups {
 		groups = append(groups, gid)
-		predMap[gid] = make([]string, 0)
+		preds := make([]string, len(group.Tablets))
```
Comment (jarifibrahim) [inline worker/backup_ee.go:196]: make([]string, 0, len(group.Tablets)

File: worker/backup_ee.go [hunk 2]
```diff
@@ -244,9 +260,6 @@ func ProcessListBackups(ctx context.Context, location string, creds *x.MinioCred
 		return nil, errors.Wrapf(err, "cannot read manifests at location %s", location)
 	}
 
-	res := make([]*Manifest, 0)
-	for _, m := range manifests {
-		res = append(res, m)
-	}
+	res := append([]*Manifest(nil), manifests...)
```
[the remark this answers is not in the document]
Comment (jarifibrahim) [reply worker/backup_ee.go:263]: I think we can remove it.

**Review by jarifibrahim — APPROVED** [review]

**Comment by danielmai** [comment]
This shouldn't be backported to the previous releases, right? This now makes the backup call asynchronous.

**Comment by ajeetdsouza** [comment]
No, this is a breaking API change, so I'd say we should restrict it to the new release.

**Pull request merged**